packages/url_launcher/url_launcher_android/example/integration_test/url_launcher_test.dart

```diff
@@ -20,4 +20,27 @@ void main() {
     // sms:, tel:, and mailto: links may not be openable on every device, so
     // aren't tested here.
   });
+
+  testWidgets('launch and close', (WidgetTester _) async {
+    final UrlLauncherPlatform launcher = UrlLauncherPlatform.instance;
+
+    // Launch a url then close.
+    expect(
+        await launcher.launch('https://flutter.dev',
+            useSafariVC: true,
+            useWebView: true,
+            enableJavaScript: false,
+            enableDomStorage: false,
+            universalLinksOnly: false,
+            headers: <String, String>{'my_header_key': 'my_header_value'}),
+        true);
+    // Allow time for page to load.
+    await Future<void>.delayed(const Duration(seconds: 3));
+    await launcher.closeWebView();
+    // Delay required to catch android side crashes in onDestroy
+    await Future<void>.delayed(const Duration(seconds: 3));
```

Comment:
I think we would need https://github.com/flutter/flutter/issues/20034 to remove this; a TODO referencing that issue wouldn't hurt.

In the meantime, do we really need 3 seconds to reliably catch closing the page? That seems crazy long.

Reply:
3s was the shortest I could make it and have the test reliability catch a crash that can happen in onDestroy on my physical phone. I ended up verifying on an emulator and actually needed to bump the delay to 5s. 3 would not reliably catch the potential crash and 4s was flakey in catching the crash. 

I think we need to wait the full 5s otherwise this test does not actually catch a regression in the code that was fixed.